AlphaWallet/RPC/Commands/web3wift-web3js/ContractERC721Transfer.swift

```diff
@@ -8,13 +8,18 @@ import BigInt
 
 struct ContractERC721Transfer: Web3Request {
 
-    //function transfer(address _to, uint256 _tokenId) external; - TODO cannot transfer by bulk, should group one by one
     typealias Response = String
-    let address: String
+    let from: String
+    let to: String
     let tokenId: String
+    let contractAddress: String
 
     var type: Web3RequestType {
-        let abi = "{\"constant\":false,\"inputs\":[{\"name\":\"_to\",\"type\":\"address\"},{\"name\":\"_tokenId\",\"type\":\"uint256\"}],\"name\":\"transfer\",\"outputs\":[],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"function\"}, [\"\(address)\", \(tokenId)]"
+        var abi = "{\"constant\":false,\"inputs\":[{\"name\":\"_to\",\"type\":\"address\"},{\"name\":\"_tokenId\",\"type\":\"uint256\"}],\"name\":\"transfer\",\"outputs\":[],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"function\"}, [\"\(to)\" \(tokenId)]"
+        let abiSafeTransferTo = "{ \"constant\": false, \"inputs\": [ { \"name\": \"_from\", \"type\": \"address\" }, { \"name\": \"_to\", \"type\": \"address\" }, { \"name\": \"_tokenId\", \"type\": \"uint256\" } ], \"name\": \"safeTransferFrom\", \"outputs\": [], \"payable\": true, \"stateMutability\": \"payable\", \"type\": \"function\"},  [\"\(from)\" \(to), \(tokenId)]"
+        if(Constants.legacy721Addresses.contains(contractAddress.drop0x.lowercased())) {
```

Comment:
Looks like the condition is wrong. Should be:

```
if !legacy.contains  {
    abi = abiSafeTransferTo
}
```

But maybe better to separate it so you can avoid the `var` and no dangling `if` like this:

```
let abiLegacyTransfer = "{\"constant\":false,\"inputs\":[{\"name\":\"_to\",\"type\":\"address\"},{\"name\":\"_tokenId\",\"type\":\"uint256\"}],\"name\":\"transfer\",\"outputs\":[],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"function\"}, [\"\(to)\" \(tokenId)]"
let abiSafeTransferFrom = "{ \"constant\": false, \"inputs\": [ { \"name\": \"_from\", \"type\": \"address\" }, { \"name\": \"_to\", \"type\": \"address\" }, { \"name\": \"_tokenId\", \"type\": \"uint256\" } ], \"name\": \"safeTransferFrom\", \"outputs\": [], \"payable\": true, \"stateMutability\": \"payable\", \"type\": \"function\"},  [\"\(from)\" \(to), \(tokenId)]"
let abi: String
if Constants.legacy721Addresses.contains(contractAddress.drop0x.lowercased()) {
    abi = abiLegacyTransfer
} else {
    abi = abiSafeTransferFrom
}
```

Also better to use `sameContract(as:)` so that it works regardless of case:

```
if Constants.legacy721Addresses.contains { $0.sameContract(as: contractAdress) }
```